rust/protocol/src/address.rs

```diff
@@ -16,7 +16,26 @@ use std::fmt;
 /// represents some user.
 ///
 /// Used in [ProtocolAddress].
-pub type DeviceId = u32;
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub struct DeviceId(pub u32);
```

Comment:
We talked about this in one of our other repos, and came to the conclusion that it was better to make the field non-`pub`, so that call sites uniformly use `into()` to go back to a u32. (As you already have here.)

Reply:
Done!